crates/ruff_linter/src/rules/ruff/rules/unnecessary_regular_expression.rs

```diff
@@ -248,3 +249,28 @@ impl<'a> ReFunc<'a> {
         })
     }
 }
+
+/// Recursively try to resolve `name` to an [`ExprStringLiteral`] in `semantic`.
+fn resolve_name<'a, 'b>(
+    name: &'a Expr,
+    semantic: &'b SemanticModel,
+) -> Option<&'b ExprStringLiteral>
+where
+    'a: 'b,
+{
```

Comment:
Nit: Do we need both lifetimes or can this be simplified to

```suggestion
fn resolve_name<'a>(
    name: &'a Expr,
    semantic: &'a SemanticModel,
) -> Option<&'a ExprStringLiteral>
{
```

Reply:
Oh yes, it can be simplified, thank you!